gossip/comm/comm_impl.go

```diff
@@ -163,7 +167,7 @@ func (c *commImpl) createConnection(endpoint string, expectedPKIID common.PKIidT
 
 	ctx, cancel = context.WithCancel(context.Background())
 	if stream, err = cl.GossipStream(ctx); err == nil {
-		connInfo, err = c.authenticateRemotePeer(stream, true)
+		connInfo, err = c.authenticateRemotePeer(stream, true, false)
 		if err == nil {
```

Comment:
Unrelated to this CR. Can be `err != nil` and return early to avoid the extra indentation.

Reply:
This is by design.
I need to close the gRPC connection, and also to cancel the gRPC stream. 
So it's not just an early return. I want cleanup operations to be at the bottom of the function so that I manage them at the same place.

Reply:
I see. Got it.